crates/config/src/toml.rs

```diff
@@ -524,7 +524,17 @@ where
         } else {
             loaded_files.insert(path_string);
         }
-        let toml_data = fs::read_to_string(config_path)?;
+        let toml_data = match fs::read_to_string(config_path) {
+            Ok(data) => data,
+            Err(err) => match err.kind() {
+                std::io::ErrorKind::NotFound => {
+                    return Err(ConfigFileError::ConfigNotFound(
+                        config_path.as_ref().to_string_lossy().into_owned(),
+                    ));
+                }
+                _ => Err(ConfigFileError::IO(err)),
+            }?,
+        };
```

Comment:
This is a decent example of a case where good old `println!` debugging is useful.

I added this just before the assertions:
```rust
println!("{}", err.to_string());
```

Output:
```txt
---- toml::tests::test_load_config_missing_include stdout ----
included config file not found: 'tests/does_not_exist.toml'
thread 'toml::tests::test_load_config_missing_include' panicked at crates/config/src/toml.rs:1128:9:
assertion failed: err.to_string().starts_with("config file not found")
```

The assertions you've written check for the entire config file being missing. That's the first error scenario. The test is configured to create the second include file missing scenario. There should be all 3 test scenarios present.

Reply:
You may need to create a new `.toml` file to support the missing plugin file test. Probably rename `missing.toml` as well to clarify that it's a missing plugin reference rather than a missing plugin file, because that will now be ambiguous.